library/src/include/utility.h

```diff
@@ -233,6 +233,40 @@ constexpr size_t rocblas_sizeof_datatype(rocblas_datatype type)
     default:                     return 0;
     }
 }
+constexpr auto rocblas_get_status_string(rocblas_status status)
+{
+    switch(status) {
+    case rocblas_status_success: return "rocblas_status_success";
+    case rocblas_status_invalid_handle: return "rocblas_status_invalid_handle";
+    case rocblas_status_not_implemented: return "rocblas_status_not_implemented";
+    case rocblas_status_invalid_pointer: return "rocblas_status_invalid_pointer";
+    case rocblas_status_invalid_size: return "rocblas_status_invalid_size";
+    case rocblas_status_memory_error: return "rocblas_status_memory_error";
+    case rocblas_status_internal_error: return "rocblas_status_internal_error";
+    case rocblas_status_perf_degraded: return "rocblas_status_perf_degraded";
+    case rocblas_status_size_query_mismatch: return "rocblas_status_size_query_mismatch";
+    case rocblas_status_size_increased: return "rocblas_status_size_increased";
+    case rocblas_status_size_unchanged: return "rocblas_status_size_unchanged";
+    default: return "rocblas_status_internal_error";
+    };
```

Comment:
Should be part of the "C" API as I needed this in examples too, hip has the equivalent in the API.   Note it exists in rocblas_test.hpp as inline const char* rocblas_status_to_string(rocblas_status status)

Reply:
See PR #772

Reply:
Change to using macro PRINT_AND_RETURN_IF_ROCBLAS_ERROR from src/include/definitions.h , and it uses new external API function rocblas_status_to_string.